### PR TITLE
stats: Extract route names with forward-slash

### PR DESCRIPTION
**Pull request by kb000**
Commit Message: Extract route names with forward-slash.
Additional Description:  RDS protocol allows almost any route name, and stats are lightly sanitized before emitting to statsd and prometheus endpoints. This change slightly relaxes the capture group for RDS route names in stats, instead of opening the floodgates to everything, because YAGNI.
Risk Level: Low
Testing: Updated unit test
Docs Changes: None
Release Notes: N/A 
Platform Specific Features: None
Fixes: #32976

### CODE REVIEW

**Review by jmarantz — APPROVED** [review]
lgtm modulo the usual concern that this might be a breaking change for someone. But I couldn't think of how it would break anything, so I'm approving.

@ggreenway wdyt?

**Comment by jmarantz** [comment]
Oh...on thing: can you take a quick look at the docs for route stats and see if they need to be updated?

**Review by ggreenway — APPROVED** [review]
LGTM

**Comment by jmarantz** [comment]
@kb100 before I merge can you just check to see if we should update doc for this syntax change?

**Comment by kb000** [comment]
> Oh...one thing: can you take a quick look at the docs for route stats and see if they need to be updated?

Oh it looks like the regex needs to be a little more relaxed. Docs say `":"` is allowed (and replaced), and it's not.
https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/rds#statistics

**Comment by kb000** [comment]
> > Oh...one thing: can you take a quick look at the docs for route stats and see if they need to be updated?
> 
> Oh it looks like the regex needs to be a little more relaxed. Docs say `":"` is allowed (and replaced), and it's not. https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/rds#statistics

Nevermind, I misread it. The `":"` never makes it into the stat name. It's sanitized out by [`sanitizeStatsName`](https://github.com/envoyproxy/envoy/blob/835ed2602107e4a39b2e0bfff63790159507d47d/source/common/stats/utility.cc#L13) well before the tag extraction code.

**Comment by kb000** [comment]
@jmarantz no doc updates needed 👍

**Pull request merged**
